src/services/Sso.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Website Toolbox Forum plugin for Craft CMS 3.x
+ * Single Sign On Cloud Based plugin for CraftCMS
+ * @link      https://websitetoolbox.com/
+ * @copyright Copyright (c) 2020 Website Toolbox
+ */
+namespace websitetoolbox\websitetoolboxforum\services;
+use websitetoolbox\websitetoolboxforum\models\Settings;
+use websitetoolbox\websitetoolboxforum\Websitetoolboxforum;
+use Craft;
+use craft\base\Component;
+use craft\web\View;
+use craft\services\Config;
+define('WT_API_URL', 'https://api.websitetoolbox.com/dev/api');
+/**
+ * @author    Website Toolbox
+ * @package   Websitetoolboxforum
+ * @since     3.0.0
+ */
+class Sso extends Component
+{    
+     function afterLogin(){  
+          $token = Craft::$app->getSession()->get(Craft::$app->getUser()->tokenParam); 
+          if($token){ 
+               $forumApiKey = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumApiKey');
+               $forumUrl    = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumUrl');
+               if($forumApiKey){ 
+                    $myUserQuery  = \craft\elements\User::find();
+                    $userEmail    = Craft::$app->getUser()->getIdentity()->email;
+                    $userId       = Craft::$app->getUser()->getIdentity()->id;
+                    $userName     = Craft::$app->getUser()->getIdentity()->username;
+                    $RequestUrl   = $forumUrl."/register/setauthtoken";
+                    $postData     = array('type'=>'json','apikey' => $forumApiKey, 'user' => $userName,'email'=>$userEmail,'externalUserid'=>$userId);
+                    $result       = Websitetoolboxforum::getInstance()->sso->sendApiRequest('POST',$RequestUrl,$postData,'json'); 
```

Comment:
Please check error as well, in case of return error then no need to set any thing.

Reply:
done